bundles/org.openhab.binding.mycroft/README.md

```diff
@@ -0,0 +1,161 @@
+|---------------|---------------------------------|-------------------------------------------------------------------------|-----------|
+| messageTypes  | List of string, comma separated | Only these message types will be forwarded to the Full Message Channel  |   No      |
+
+
+## Full Example
+
+A manual setup through a `things/mycroft.things` file could look like this:
+
+```java
+Thing mycroft:mycroft:myMycroft "Mycroft A.I." @ "Living Room" [host="192.168.X.X", port=8181] { 
+    Channels:
+        Type full-message-channel : Text [
+            messageTypes="message.type.1,message.type.4"
+        ]
+}
+```
+
+### Item Configuration
+
+The `mycroft.item` file :
+
+```java
+Switch myMycroft_mute                  "Mute"                      { channel="mycroft:mycroft:myMycroft:volume_mute" }
+Switch myMycroft_duck                  "Duck"                      { channel="mycroft:mycroft:myMycroft:volume_duck" }
+Dimmer myMycroft_volume                "Volume [%d]"               { channel="mycroft:mycroft:myMycroft:volume" }
+Player myMycroft_player                "Control"                   { channel="mycroft:mycroft:myMycroft:player" }
+Switch myMycroft_listen                "Wake and listen"           { channel="mycroft:mycroft:myMycroft:listen" }
+String myMycroft_speak                 "Speak STT"                 { channel="mycroft:mycroft:myMycroft:speak" }
+String myMycroft_utterance             "Utterance"                 { channel="mycroft:mycroft:myMycroft:utterance" }
+String myMycroft_fullmessage           "Full JSON message"         { channel="mycroft:mycroft:myMycroft:full_message" }
+```
+
+### Sitemap Configuration
+
+A `demo.sitemap` file :
+
+```perl
+sitemap demo label="myMycroft"
+{
+    Frame label="myMycroft" {
+        Switch    item=myMycroft_mute
+        Switch    item=myMycroft_duck
+        Slider    item=myMycroft_volume
+        Default   item=myMycroft_player
+        Switch    item=myMycroft_listen
+        Text      item=myMycroft_speak
+        Text      item=myMycroft_utterance
+        Text      item=myMycroft_fullmessage
+    }
+}
+```
+
+## Rule Actions
+
+This binding includes a number of rule actions, which allow triggering Mycroft from within rules.
+
+
+```
+(Rule DSL)
+val mycroftAction = getActions("mycroft","mycroft:mycroft:<uid>")
+```
+
+```
+(javascript JSR)
+var mycroftAction = actions.get("mycroft", "mycroft:mycroft:<uid>");
+```
+
+
+Where uid is the Thing UID of the Mycroft thing.
+
+Once this action instance is retrieved, you can invoke the `speak' method on it:
+
+```
+mycroftAction.speak("Hello world!")
+```
+
+
+The following actions are supported.
+
+
+| Action                     | Description  |
+|----------------------------|--------------|
+| speak(String message) | Mycroft will say this. |
+| listen()          | Trigger Mycroft as if the wake word was detected |
+| utterance(String utterance) | Ask Mycroft something|
```

Comment:
I think this action is the correct way of implementing this functionality, but then I would also remove the `utterance` channel, because its not needed.

Reply:
Yes, speak / utterance has the exact same issues.